File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0004-lending/ApplicationRiskScoreModelRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "ApplicationRiskScoreModel"})
public class ApplicationRiskScoreModelRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private java.math.BigDecimal applicationRiskScoreModel;

    public ApplicationRiskScoreModelRuleOutput(boolean matched) {
        super(matched);
    }

    public java.math.BigDecimal getApplicationRiskScoreModel() {
        return this.applicationRiskScoreModel;
    }
    public void setApplicationRiskScoreModel(java.math.BigDecimal applicationRiskScoreModel) {
        this.applicationRiskScoreModel = applicationRiskScoreModel;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ApplicationRiskScoreModelRuleOutput other = (ApplicationRiskScoreModelRuleOutput) o;
        if (this.getApplicationRiskScoreModel() != null ? !this.getApplicationRiskScoreModel().equals(other.getApplicationRiskScoreModel()) : other.getApplicationRiskScoreModel() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getApplicationRiskScoreModel() != null ? this.getApplicationRiskScoreModel().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", applicationRiskScoreModel='%s'", applicationRiskScoreModel));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-test-cases/proto/1.1/0004-lending/translator/expected/proto3/kotlin/dmn/ApplicationRiskScoreModelRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "ApplicationRiskScoreModel"])
class ApplicationRiskScoreModelRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("ApplicationRiskScoreModel")
    var applicationRiskScoreModel: java.math.BigDecimal? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as ApplicationRiskScoreModelRuleOutput
        if (if (this.applicationRiskScoreModel != null) this.applicationRiskScoreModel != other.applicationRiskScoreModel else other.applicationRiskScoreModel != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.applicationRiskScoreModel != null) this.applicationRiskScoreModel.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", applicationRiskScoreModel='%s'", applicationRiskScoreModel))
        result_.append(")")
        return result_.toString()
    }
}
